### PR TITLE
🌟 #1960 Implement users list search in frontend

### PR DESCRIPTION
**Pull request by stephanevieira75**

### CODE REVIEW

File: twake/frontend/src/app/features/users/types/user.ts [hunk 1]
```diff
@@ -41,9 +49,9 @@ export type UserType = {
   timezone_offset?: string;
   tutorial_status?: any;
   username: string;
-  companies?: { company: CompanyType; role: CompanyRoleType; status: CompanyStatusType }[];
+  companies?: UserCompanyType[];
   preferences: UserPreferencesType;
-  workspaces?: WorkspaceType[];
+  workspaces?: UserWorkspaceType[];
```
Comment (RomaricMourgues) [inline twake/frontend/src/app/features/users/types/user.ts:54]: Here we should add some comments to precise that this fields are filled when available and so we cannot rely on it except on search service for filtering

Comment (RomaricMourgues) [inline twake/frontend/src/app/features/users/types/user.ts:54]: Here we should add some comments to precise that this fields are filled when available and so we cannot rely on it except on search service for filtering

File: twake/frontend/src/app/features/channel-members/hooks/use-channel-guests.ts [hunk 2]
```diff
@@ -31,6 +34,11 @@ export const useChannelGuests = (
     );
 
     if (channelGuestsUpdated) _setChannelGuests(channelGuestsUpdated);
+
+    const usersIdx = channelGuestsUpdated.map(guest => guest.user_id || '');
+    const users = await UserAPIClient.list(usersIdx, [companyId]);
```
Comment (RomaricMourgues) [inline twake/frontend/src/app/features/channel-members/hooks/use-channel-guests.ts:39]: Not sure about doing another call to backend here could we put a Fixme comment ?

File: twake/frontend/src/app/features/global/services/search-list-manager-service.ts [hunk 3]
```diff
@@ -35,8 +35,9 @@ class SearchListManager extends Observable {
```
Comment (RomaricMourgues) [inline twake/frontend/src/app/features/global/services/search-list-manager-service.ts:35]: So what is this file used for now ?

File: twake/frontend/src/app/views/applications/messages/parts/thread-avatar.tsx [hunk 4]
```diff
@@ -24,7 +24,7 @@ export default (props: Props) => {
   let { message } = useMessage(context);
 
   let user =
-    useRecoilValueLoadable(UsersState(message.user_id)).valueMaybe() ||
+    useRecoilValueLoadable(UserSelector(message.user_id)).valueMaybe() ||
```
Comment (RomaricMourgues) [inline twake/frontend/src/app/views/applications/messages/parts/thread-avatar.tsx:27]: Can't we use useUser here or am I missing something?

Comment (RomaricMourgues) [inline twake/frontend/src/app/views/applications/messages/parts/thread-avatar.tsx:27]: Can't we use useUser here or am I missing something?

File: twake/frontend/src/app/views/client/main-view/MainView.tsx [hunk 5]
```diff
@@ -23,6 +24,8 @@ const MainView: FC<PropsType> = ({ className }) => {
   const workspaceId = useRouterWorkspace();
   const channelId = useRouterChannel();
 
+  useUserList();
```
Comment (RomaricMourgues) [inline twake/frontend/src/app/views/client/main-view/MainView.tsx:27]: Could we add a comment here to specify why we do this ?

Comment (RomaricMourgues) [inline twake/frontend/src/app/views/client/main-view/MainView.tsx:27]: Could we add a comment here to specify why we do this ?

**Review by RomaricMourgues — CHANGES_REQUESTED** [review]
Good work, mostly comments to add before I can merge it :)

File: twake/frontend/src/app/features/global/services/search-list-manager-service.ts (hunk 3 above)
Comment (stephanevieira75) [reply twake/frontend/src/app/features/global/services/search-list-manager-service.ts:35]: `WorkspaceChannelList.tsx` and `SearchListContainer.tsx` for cmdOrCtrl+K.

We basically use it to transform data into generic channels

File: twake/frontend/src/app/features/users/types/user.ts (hunk 1 above)
Comment (stephanevieira75) [reply twake/frontend/src/app/features/users/types/user.ts:57]: done

File: twake/frontend/src/app/features/channel-members/hooks/use-channel-guests.ts (hunk 2 above)
Comment (stephanevieira75) [reply twake/frontend/src/app/features/channel-members/hooks/use-channel-guests.ts:40]: done

File: twake/frontend/src/app/views/applications/messages/parts/thread-avatar.tsx (hunk 4 above)
Comment (stephanevieira75) [reply twake/frontend/src/app/views/applications/messages/parts/thread-avatar.tsx]: Updated 👍🏻

Comment (stephanevieira75) [reply twake/frontend/src/app/views/applications/messages/parts/thread-avatar.tsx]: Done

File: twake/frontend/src/app/views/client/main-view/MainView.tsx (hunk 5 above)
Comment (stephanevieira75) [reply twake/frontend/src/app/views/client/main-view/MainView.tsx:28]: Done

**Pull request merged**